lib/bolt/node/winrm.rb

```diff
@@ -428,6 +426,12 @@ def process_from_extension(path)
           'puppet.bat',
           ['apply', "\"#{path}\""]
         ]
+      else
+        # Run the script via cmd, letting Windows extension handling determine how
+        [
+          'cmd.exe',
+          ['/c', "\"#{path}\""]
```

Comment:
This may not work how you think if the default action associated with the file extension is something like `Edit` or `Open in Editor` rather than something like `Run` or `Execute`.

We intentionally chose not to go this route and to stick to a whitelist because you have no control over how the file associations are setup, and could end up popping a Window'd UI rather than actually running code through an appropriate interpreter.

Reply:
Also - for the general approach, I think this was getting kicked down the road until users could register their own extensions inside the Bolt config.

Reply:
That is a concern, but I'm not sure how a whitelist helps with that. This also prevents Bolt from running tasks that would be valid in PE.

If we want it to be node-specific, Windows already has one: PATHEXT.

If we made it configurable, users would be able to specify other things to try that might misbehave as you say. I'm not sure how doing that via an extra config option is terribly different than doing it by selecting something with a specific file extension.

If it's not user configurable, then we simply prevent users from doing something they want to do.

Reply:
Yes, [BOLT-146](https://tickets.puppetlabs.com/browse/BOLT-146). But that ticket is mostly targeting other more complicated scenarios.

I plan to look at BOLT-146 soon. If we don't believe this is desirable without BOLT-146, I can focus on that instead.

Reply:
Yeah, totally understand that we want users to be able to have flexibility in what they run.... no disagreement there.

I'm just not sure this is the way to go about it. For instance, create an empty text file on disk ending in `.txt`. From a terminal run `cmd.exe /c foo.txt`

Reply:
Ah, you're right about `PATHEXT`. I think I assumed it was a problem rather than how Bolt was treating files with extensions it didn't recognize (if it didn't recognize the extension, it would try to run it as a .ps1 file).

Reply:
Yeah, I'm lukewarm on treating extensionless files as PowerShell, because extensionless files are foreign to Windows users / potentially confusing... so wouldn't be totally upset if extensions were made mandatory